### DISCUSSION TITLE
BaseDeployment-SingleMachine post deployment

### DISCUSSION THREAD

**Discussion by wmedernach** [Esri/arcgis-powershell-dsc#584] [Q&A]
Is there a wiki that describes any required post deployment steps?

I've just ran the BaseDeployment-SingleMachine and with the exception of the Portal patches failing because the _Portal for ArcGIS 10.9.1 Validation and Repair tool_ needed to be run first, everything else seemed to install fine with no other errors.

The problem seems to be that none of the post-install configuration was completed.  No Site was created, the WA's did not get configured, which I suppose could be a result of no Site being created, so I'm just curious what I'm missing?

**Comment by cameronkroeker** [comment]
Hi @wmedernach, sounds like you may have only ran the `Invoke-ArcGISConfiguration` command with `-Mode Install`. The create site, web adaptor registration, ssl certificate imports, etc happen when using `-Mode InstallLicenseConfigure`.

- https://github.com/Esri/arcgis-powershell-dsc/wiki/PowerShell-DSC-ArcGIS-Module-Command-Line-Parameters#-mode
- https://github.com/Esri/arcgis-powershell-dsc/wiki/Create-a-base-ArcGIS-Enterprise-deployment-on-one-machine
- https://github.com/Esri/arcgis-powershell-dsc/blob/main/SampleConfigs/v4/v4.4.0/Base%20Deployment/BaseDeployment-SingleMachine.json
- https://github.com/Esri/arcgis-powershell-dsc/wiki/v4.4.0-Variables-reference-page-for-JSON-configuration-files

**Reply by wmedernach** [reply]
Thanks again @cameronkroeker.

I've been able to get further with the install and configuration, with the exception of the Web Adaptor config.  It is erroring out with:
_Trace-DSCJob : 11/26/2024 5:29:32 PM: PowerShell DSC resource ArcGIS_WebAdaptor  failed to execute Set-TargetResource functionality with error message: [ERROR]:- Web Adaptor configuration failed. Output - Portal configuration requires fully qualified domain name in the WebAdaptor URL_

I have set the FQDN under the SslCertificates section, but it doesn't seem to be using it based on the log.

This is the AllNodes section:

```
"AllNodes": [
	{
		"NodeName": "machinename",
		"DataStoreTypes": [
			"Relational"
		],
		"Role": [
			"Server",
			"Portal",
			"DataStore",
			"WebAdaptor"
		],
		"WebAdaptorConfig":[
			{
				"Role": "Portal",
				"Context": "portal",
				"AdminAccessEnabled": true
			},
			{
				"Role": "Server",
				"Context": "arcgis",
				"AdminAccessEnabled": true
			}
		],
		"SslCertificates": [
			{
				"CNameFQDN": "machinename.rg.dom.ad",
				"Target": [
					"Server", "Portal"
				]
			}
		]
	}
]
```

Is there something else I'm missing?

**Reply by cameronkroeker** [reply]
I suspect the Get-FQDN command is not returning an FQDN. Can you run these 2 commands and see if a FQDN is returned?

```powershell
Get-FQDN machinename
```

and

```powershell
Get-FQDN $env:COMPUTERNAME
```

**Reply by wmedernach** [reply]
Both
```
Get-FQDN machinename
Get-FQDN $env:COMPUTERNAME
```

just return the machinename.

Fyi, if I run [System.Net.Dns]::GetHostByName(($env:computerName)).HostName, the FQDN is returned.

**Reply by cameronkroeker** [reply]
Thanks, I think that could be the issue. Which version of the ArcGIS module are you using?

**Reply by wmedernach** [reply]
I did not clone the repo locally and just downloaded and installed the zip latest last week, so I'm assuming that's v4.4.0 based on the release notes?

**Reply by cameronkroeker** [reply]
Yes, if it was last week then its v4.4.0. Try setting the Web Adaptor FQDN using this variable:

```json
"WebAdaptorConfig":[
   {
      "Role": "Portal",
      "Context": "portal",
      "AdminAccessEnabled": true,
      "HostName": "webadaptorNodeHostname.domain.com"
   },
   {
      "Role": "Server",
      "Context": "arcgis",
      "AdminAccessEnabled": true,
      "HostName": "webadaptorNodeHostname.domain.com"
    }
],
```

This should override the default value of the web adaptor url being retrieved by the ArcGIS Module to register with portal and server.

**Reply by wmedernach** [reply]
Thanks @cameronkroeker, unfortunately that still didn't work either, even though I can successfully access the URL shown in the log and it brings up the Web Adaptor config page and allows me to manually register the WA with that URL and admin ports.

11/27/2024 8:28:36 PM: [MACHINENAME]:  [[ArcGIS_WebAdaptor]ConfigurePortalWebAdaptormachinename-portal] https://machine.na.abc.ad/portal/webadaptor

**Reply by cameronkroeker** [reply]
@wmedernach Let's see if we can print all the values the ArcGIS Module is passing in for the web adaptor configuration to see if they are valid. Curious to see if both the WA Url and Site Url are being passed in as an FQDN.

Add `Write-Verbose $Arguments` here:

https://github.com/Esri/arcgis-powershell-dsc/blob/539a5adada7de60e3b5ce801a74020c0e3112d61/Modules/ArcGIS/DSCResources/ArcGIS_WebAdaptor/ArcGIS_WebAdaptor.psm1#L479-L481

For example:
```
}else{
   $Arguments = "/m portal /w $WAUrl /g $SiteURL /u $($SiteAdministrator.UserName) /p `"$($SiteAdministrator.GetNetworkCredential().Password)`""
   Write-Verbose $Arguments
}
```

After saving the file, ensure to launch a fresh Powershell window and re-run the `Invoke-ArcGISConfiguration` command. If you pass in `-DebugSwitch` with the `Invoke-ArcGISConfiguration` command it will display it in the powershell window the argument list, however keep in mind this will also enable debug logging on portal and server as well. Without the `DebugSwitch` the output will be in the ArcGIS Module DSC log file.

**Reply by wmedernach** [reply]
This is the output:

[[ArcGIS_WebAdaptor]ConfigurePortalWebAdaptormachinename-portal] /m portal /w https://machinename.na.abc.ad/portal/webadaptor /g https://machinename:7443 /u username /p "password"

It seems like these 2 URLS reversed? 
Shouldn't it be calling the WA config via `https://machinename/portal/webadaptor` and using `https://machinename.na.abc.ad:7443` to register the WA?

**Reply by cameronkroeker** [reply]
The web adaptor url (/w) looks correct, however, the portal url (/g) is not correct. It also needs to be a FQDN, which is likely the culprit of the error. I suspect this is because the output of the Get-FQDN function is retrieving the hostname rather than the FQDN.

Can you try specifying the FQDN for the NodeName in the json configuration file to see if this works?

```json
"NodeName": "machinename.na.abc.ad"
```

**Reply by wmedernach** [reply]
Thanks again @cameronkroeker, I've been able to successfully deploy the hosting machine and register the WA's by specifying the FQDN in the NodeName.  The only thing I did note in some initial poking around is that, in the server admin home > machines, when I click on the machine name it throughs a 404.7 http error:

```
HTTP Error 404.7 - Not Found
The request filtering module is configured to deny the file extension.
```

So I'm not exactly sure why that is?

**Reply by cameronkroeker** [reply]
Hi @wmedernach, I have not seen this before but I believe the request filtering module is apart of IIS. I suggest starting with IIS to see if there are any request filtering rules configure/defined or settings that could be blocking file extensions.

**Reply by cameronkroeker** [reply]
Actually @wmedernach I stumbled upon this kb article. Even though its for portal I suspect it can also apply to arcgis server as well:
https://support.esri.com/en-us/knowledge-base/problem-unable-to-connect-to-basic-functionality-in-por-000021912

**Reply by wmedernach** [reply]
Thanks once again @cameronkroeker!  The '.ad' in the server name was indeed being seen as a file and being filtered out.  I think we can put this post to bed now and I can proceed onto the Image Server DSC deployment.
